zcash_client_sqlite/src/wallet.rs

```diff
@@ -4188,120 +4140,106 @@ pub(crate) fn queue_tx_retrieval(
     )?;
 
     for txid in txids {
-        let tx_expiry = q_tx_expiry
-            .query_row(
-                named_params! {
-                    ":txid": txid.as_ref(),
-                },
-                |row| row.get::<_, Option<u32>>(0),
-            )
-            .optional()?
-            .flatten();
-
         stmt_insert_tx.execute(named_params! {
             ":txid": txid.as_ref(),
             ":status_type": TxQueryType::Status.code(),
             ":enhancement_type": TxQueryType::Enhancement.code(),
             ":dependent_transaction_id": dependent_tx_ref.map(|r| r.0),
-            ":request_expiry": tx_retrieval_expiry(chain_tip, tx_expiry)
         })?;
     }
 
     Ok(())
 }
 
-// Computes a block height at which transaction status requests should be considered
-// expired. Note that even expired requests will be serviced; expiry is only used
-// to determine when a request can be deleted and not retried.
-//
-// TODO: the underlying `request_expiry` mechanism that this method is being used to configure is
-// not sufficiently precise and should probably be scrapped in favor of something better; however,
-// this will do for now.
-fn tx_retrieval_expiry(chain_tip_height: BlockHeight, tx_expiry: Option<u32>) -> u32 {
-    match tx_expiry {
-        // If the transaction never expires or we don't know its expiry, keep checking for the next
-        // expiry delta from the chain tip.
-        Some(0) | None => u32::from(chain_tip_height) + DEFAULT_TX_EXPIRY_DELTA,
-        // If the transaction expires in the future, the request can expire with the transaction.
-        Some(h) if h > u32::from(chain_tip_height) => h + VERIFY_LOOKAHEAD,
-        // The transaction has already expired; we can keep checking for a few blocks, but if we
-        // don't get a positive response for the transaction being mined soon, we can assume it
-        // will never be mined.
-        Some(_) => u32::from(chain_tip_height) + VERIFY_LOOKAHEAD,
-    }
-}
-
 /// Returns the vector of [`TransactionDataRequest`]s that represents the information needed by the
 /// wallet backend in order to be able to present a complete view of wallet history and memo data.
 pub(crate) fn transaction_data_requests(
     conn: &rusqlite::Connection,
 ) -> Result<Vec<TransactionDataRequest>, SqliteClientError> {
-    let mut tx_retrieval_stmt =
-        conn.prepare_cached("SELECT txid, query_type FROM tx_retrieval_queue")?;
+    // We will return both explicitly constructed status requests, and a status request for each
+    // transaction that is known to the wallet for which we don't have the mined height,
+    // and for which we have no positive confirmation that the transaction expired unmined.
+    //
+    // For transactions with a known expiry height of 0, we will continue to query indefinitely.
+    // Such transactions should be rebroadcast by the wallet until they are mined.
```

Comment:
```suggestion
    // Such transactions should be rebroadcast by the wallet until they are either mined or
    // conflict with another mined transaction.
```
which implies we also need to ensure that we stop querying once we observe the transaction is conflicted (i.e. the junction tables tell us that it spends notes or coins that are also spent in some other tx that is mined), and that we _start_ querying again if that conflicting tx itself gets unmined in a reorg.